test/framework/test_composite.py

```diff
@@ -1000,3 +1017,127 @@ def circuits(self):
+
+        # Set a batch experiment
+        batch_exp = BatchExperiment([par_exp1, par_exp2], flatten_results=True)
+        exp_data = batch_exp.run(backend)
+
+        self.assertExperimentDone(exp_data)
+
+    def test_new_workflow_all_same(self):
+
+        data = [
+            {
+                "metadata": {
+                    "experiment_type": "BatchExperiment",
+                    "composite_metadata": [
+                        {
+                            "experiment_type": "ParallelExperiment",
+                            "composite_index": [0, 1],
+                            "composite_metadata": [{}, {}],
+                            "composite_qubits": [[0], [1]],
+                            "composite_clbits": [[], []],
+                        }
+                    ],
+                    "composite_index": [0],
+                }
+            },
+            {
+                "metadata": {
+                    "experiment_type": "BatchExperiment",
+                    "composite_metadata": [
+                        {
+                            "experiment_type": "ParallelExperiment",
+                            "composite_index": [0, 1],
+                            "composite_metadata": [{}, {}],
+                            "composite_qubits": [[0], [1]],
+                            "composite_clbits": [[], []],
+                        }
+                    ],
+                    "composite_index": [1],
+                },
+            }
+        ]
+        
+        
+        backend = FakeBackend()
+
+        exp1 = FakeExperiment([0])
+        exp2 = FakeExperiment([1])
+        par_exp1 = ParallelExperiment([exp1, exp2], flatten_results=True)
+
+        exp3 = FakeExperiment([0])
+        exp4 = FakeExperiment([1])
+        par_exp2 = ParallelExperiment([exp3, exp4], flatten_results=True)
+
+        # Set a batch experiment
+        batch_exp = BatchExperiment([par_exp1, par_exp2], flatten_results=True)
+        exp_data = batch_exp.run(backend)
+        
+        for metadata,datum in zip(data,exp_data.data()):
+            self.assertTrue(metadata in datum)
+        
+    
+    def test_new_workflow_child_count(self):
+        
+        backend = FakeBackend()
+
+        exp1 = FakeExperiment([0])
+        exp2 = FakeExperiment([1])
+        par_exp1 = ParallelExperiment([exp1, exp2], flatten_results=True)
+
+        exp3 = FakeExperiment([0])
+        exp4 = FakeExperiment([1])
+        par_exp2 = ParallelExperiment([exp3, exp4], flatten_results=True)
+
+        # Set a batch experiment
+        batch_exp = BatchExperiment([par_exp1, par_exp2], flatten_results=True)
+        exp_data = batch_exp.run(backend)
+        
+        for child in exp_data.child_data():
+
+            self.assertEqual(len(child.child_data()),2)
+            
+            for inner_child in child.child_data():
+                
+                self.assertEqual(len(inner_child.child_data()),0)
```

Comment:
Since `FakeBackend` has hard-coded results (i.e. it returns `counts: {"0": 100}` regardless of input circuit), it's worth checking all children of inner_child returns circuit results of count "0": 100.
https://github.com/Musa-Sina-Ertugrul/qiskit-experiments/blob/990351344a6c3ff2b74663a3a3f6d020a1c9d4bb/qiskit_experiments/test/fake_backend.py#L65-L87